src/utils/hooks/api/use-get-collateral-currencies-data.tsx

```diff
@@ -0,0 +1,61 @@
+import { CollateralCurrencyExt, CurrencyExt } from '@interlay/interbtc-api';
+import { Currency, ExchangeRate } from '@interlay/monetary-js';
+import Big from 'big.js';
+import { useQuery, UseQueryResult } from 'react-query';
+
+import { useGetCollateralCurrencies } from './use-get-collateral-currencies';
+
+type CollateralCurrenciesData = { [currencyTicker: string]: CollateralCurrencyData };
+
+type CollateralCurrencyData = {
+  liquidationThreshold: Big;
+  secureThreshold: Big;
+  exchangeRate: ExchangeRate<Currency, CurrencyExt>;
+};
+
+const getCollateralCurrencyData = async (collateralToken: CollateralCurrencyExt): Promise<CollateralCurrencyData> => {
+  try {
+    const [liquidationThreshold, secureThreshold, exchangeRate] = await Promise.all([
+      window.bridge.vaults.getLiquidationCollateralThreshold(collateralToken),
+      window.bridge.vaults.getSecureCollateralThreshold(collateralToken),
+      window.bridge.oracle.getExchangeRate(collateralToken)
+    ]);
+
+    return { liquidationThreshold, secureThreshold, exchangeRate };
+  } catch (error) {
+    throw new Error(
+      `useGetCollateralCurrenciesData: error getting data for currency ${collateralToken.ticker}: ${error.message}`
+    );
+  }
+};
+
+const getCollateralCurrenciesData = (
```

Comment:
Did you consider using `useQueries` rather than having `useQuery` call this function?

Reply:
I think `Promise.all` is better solution in this case because queried data are atomic - all promises have to be resolved before the returned data can be used.

Reply:
Ok, sounds good :)